src/update_agent/mod.rs

```diff
@@ -136,9 +136,9 @@ impl UpdateAgentState {
 
     /// Record a failed deploy attempt in UpdateAvailable state.
     ///
-    /// This returns whether a persistent deploy failure was detected
-    /// and the target update abandoned.
-    fn record_failed_deploy(&mut self) -> bool {
+    /// This returns the total number of failed deployment attempts
+    /// including the newly recorded failed attempt.
+    fn record_failed_deploy(&mut self) -> u8 {
```

Comment:
Hmm, I guess the function could return a tuple of `(bool, u8)` too so you don't have to redo the comparison in the callee.

Reply:
oh right, I forgot that was possible :)